### ISSUE TITLE
Create separate routes for sign up vs. log in

### ISSUE THREAD

**Issue by michael-tharrington** [forem/forem#9902]
**Is your feature request related to a problem? Please describe.**

Occasionally a user tries to log into a DEV account and they mistakenly create a second DEV account. This problem is best described in [this article](https://codeburst.io/how-to-do-it-wrong-third-party-auth-aka-social-login-d7f24abecef5) under the heading "Signing up users who are trying to log in".

**Describe the solution you'd like**

Create separate routes for signing up vs. logging in.

**Additional context**

A DEV user wrote in to point this out.

Also, there's a related solution to this issue here - https://github.com/forem/forem/issues/5010 that could be great to enact. But, I still think we should separate our signing up and logging in flows.

**Comment by boardfish** [comment]
If an account doesn't yet exist against the given GitHub/Twitter account after the OAuth callback redirects back to Forem, maybe we should show a screen that asks the user to confirm whether they'd like to create an account. They could then also choose to sign in with the other service or use password auth, in case they already have creds for those.

**Comment by rhymes** [comment]
@lisasy does this issue still apply now that you've been working on refactoring the authentication/login process?

**Comment by lisasy** [comment]
@rhymes Yes, it seems it has been fixed with something that Ben helped out on.

Login path: `https://dev.to/enter`
Create account path: `https://dev.to/enter?state=new-user`

**Comment by boardfish** [comment]
Yeah, that's roughly what I envisioned. I recognized that there were two separate views for login, one with a sign-up form and one without, so this fix would be to cater for those that already have accounts (exclusively against OAuth providers).

**Comment by cmgorton** [comment]
@fdoxyz @michael-tharrington should we label this as a potential RFC?

**Comment by michael-tharrington** [comment]
I think we could absolutely take another look at making the sign up/sign in pages a bit more clear! Someone recently wrote in and was confused why they couldn't sign up with email after they noticed that they could sign in with email:

![Image 2021-03-01 at 3 05 32 PM](https://user-images.githubusercontent.com/16007075/109552524-e11b8d00-7a9f-11eb-9b74-617e80f9b90d.png)

I think stage 0 RFC makes a lot of sense here as there are quite a few ways we could try to solve this one!

**Comment by mikeydorje** [comment]
Based on code review, the core feature requested here was implemented in 2020 as noted in the October 2020 comments. The codebase has distinct authentication flows using the `state` parameter:

While March 2021 comments discussed potential additional UX improvements, this issue has had no activity since then and is now over 4 years old. Closing this as stale and the original request as implemented.

If there are still UX concerns around the sign-up/sign-in flow, please open a new issue with specific examples and reproduction steps against the current codebase.
